cmd/cluster-init/main.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+type options struct {
+	clusterName string
+	releaseRepo string
+}
+
+func (o options) String() string {
+	return fmt.Sprintf("cluster-name: %s\nrelease-repo: %s",
+		o.clusterName,
+		o.releaseRepo)
+}
+
+func parseOptions() options {
+	var o options
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.clusterName, "cluster-name", "", "The name of the new cluster.")
+	fs.StringVar(&o.releaseRepo, "release-repo", "", "Path to the root of the openshift/release repository.")
+
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("cannot parse args: ", os.Args[1:])
+	}
+	return o
+}
+
+func validateOptions(o options) []error {
+	var errs []error
+	if o.clusterName == "" {
+		errs = append(errs, fmt.Errorf("--cluster-name must be provided"))
+	} else if strings.ContainsAny(o.clusterName, " \t\n") {
```

Comment:
You might try 

```go
else if trimmed := strings.TrimFunc(o.clusterName, unicode.IsSpace); trimmed != o.clusterName {
```

Reply:
It appears that only trims out the leading and trailing spaces. We also need to exclude interior spaces.

Reply:
OK, still please use `unicode.IsSpace` in your check - there's a whole lot more to whitespace than your three chars :)

Reply:
Yes, I figured this was a very naive approach for checking for whitespace. I'll work the `unicode.IsSpace` in there. Thanks!